### PR TITLE
Running test suite as part of PyPI release #147

### PR DESCRIPTION
**Pull request by tkoyama010**
Running test suite as part of PyPI release #147
This is a sprint contribution in ScipyJapan2020 (#163) .

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -22,7 +25,11 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install setuptools wheel twine
+    - name: Test
+      run: |
+        make test
     - name: Build and publish
+      if: contains(contains(github.ref, 'master')
```
Comment (tkoyama010) [inline .github/workflows/python-publish.yml:32]:
```suggestion
      if: contains(github.ref, 'master')
```

**Comment by tkoyama010** [comment]
Oh my. Test script which we created in #172 seems not working :cry:
https://github.com/dask/dask-image/pull/175/checks?check_run_id=1340168042

**Comment by tkoyama010** [comment]
Before reviewing this Pull Request, please review see #176 :bow:

File: .github/workflows/python-publish.yml [hunk 2]
```diff
@@ -6,6 +6,9 @@ name: Upload Python Package
 on:
   release:
     types: [created]
+  pull_request:
+    branches:
+      - master
```
Comment (tkoyama010) [inline .github/workflows/python-publish.yml:11]:
```suggestion
```

File: .github/workflows/python-publish.yml [hunk 3]
```diff
@@ -22,7 +25,11 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install setuptools wheel twine
+    - name: Test
+      run: |
+        make test
     - name: Build and publish
+      if: contains(github.ref, 'master')
```
Comment (tkoyama010) [inline .github/workflows/python-publish.yml:32]:
```suggestion
```

**Review by tkoyama010 — COMMENTED** [review]
:wastebasket: Remove patch of check

**Comment by tkoyama010** [comment]
Now it is ready for review :rocket:

File: .github/workflows/python-publish.yml [hunk 4]
```diff
@@ -22,7 +25,11 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install setuptools wheel twine
+    - name: Test
+      run: |
+        make test
```
Comment (GenevieveBuckley) [inline .github/workflows/python-publish.yml:27]: `make test` seems equivalent to `python setup.py test`, which is now deprecated(https://github.com/pypa/setuptools/issues/1684). 

I get a warning when I run this, see: https://github.com/GenevieveBuckley/dask-image/runs/1340523683?check_suite_focus=true

```
Run make test
python setup.py test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
```

If possible, I'd like us to replace this with something that is less likely to stop working at some future point in time.

File: .github/workflows/python-publish.yml (hunk 3 above)
Comment (GenevieveBuckley) [inline .github/workflows/python-publish.yml]: I think we can remove this line, the release tag is already attached to the master branch.

File: .github/workflows/python-publish.yml [hunk 5]
```diff
@@ -6,6 +6,9 @@ name: Upload Python Package
 on:
   release:
     types: [created]
+  pull_request:
```
Comment (GenevieveBuckley) [inline .github/workflows/python-publish.yml]: This should be:
```on:
  release:
    types: [created]
```

Not this:
```
on:
  release:
    types: [created]
  pull_request:
    branches:
      - master
```

Comment (GenevieveBuckley) [reply .github/workflows/python-publish.yml]: I understand you may want to change this to test the github action is working on your branch before we merge this, of course!

**Review by GenevieveBuckley — CHANGES_REQUESTED** [review]
Thanks for making this PR @tkoyama010 

I think we should find an alternative to `make test` / `python setup.py test` since that command is now deprecated (https://github.com/pypa/setuptools/issues/1684). Other than that, I like this a lot 😄

File: .github/workflows/python-publish.yml (hunk 5 above)
Comment (tkoyama010) [reply .github/workflows/python-publish.yml]: Yes I will remove this when we will merge this PR. Thanks for comment.

File: .github/workflows/python-publish.yml (hunk 3 above)
Comment (tkoyama010) [reply .github/workflows/python-publish.yml]: You are right. Thanks for comment.

File: .github/workflows/python-publish.yml (hunk 4 above)
Comment (tkoyama010) [reply .github/workflows/python-publish.yml:27]: Thanks. I am +1 for this.

Comment (tkoyama010) [reply .github/workflows/python-publish.yml:27]:
```suggestion
        python setup.py test
```

**Comment by tkoyama010** [comment]
Thanks @GenevieveBuckley for your review.

**Comment by GenevieveBuckley** [comment]
I'm going to close this PR in favour of https://github.com/dask/dask-image/pull/291 
The advantage with that is:
* there will be no difference between the regular test CI, and the tests that are run just before the new release is uploaded to PyPI
* it will also test all operating systems (Windows, Mac, and Linux) instead of just on Linux

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

**Comment by GenevieveBuckley** [comment]
Thanks again for all your help @tkoyama010!

**Pull request closed without merging**